src/tests/java/picard/sam/FilterSamReadsTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * The MIT License
+ *
+ * Pierre Lindenbaum - Institut du Thorax - 2016
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package picard.sam;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import picard.cmdline.CommandLineProgramTest;
+
+import java.io.File;
+import java.io.FileFilter;
+
+public class FilterSamReadsTest extends CommandLineProgramTest {
+    @Override
+    public String getCommandLineProgramName() {
+        return FilterSamReads.class.getSimpleName();
+    }
+
+    /**
+     * filters a SAM using the available javascript filters
+     */
+    @Test
+    public void testJavaScriptFilters() throws Exception {
+        // input as SAM file 
+        final File inputSam = new File("testdata/picard/sam/aligned.sam");
+        //loop over javascript filters
+        for(final File javascriptFile: new File("testdata/picard/sam/FilterSamReads").listFiles(new FileFilter() {
```

Comment:
This auto discovery is cool, I think it would be better to just hardcode the names though.  This would be a good place to use a dataprovider.  It would be good to test if the expected number of reads were actually filtered out as well.